nautobot/core/ui/object_detail.py

```diff
@@ -754,13 +780,18 @@ def get_data(self, context):
                 if field.name in ("id", "created", "last_updated"):
                     # Handled elsewhere in the detail view
                     continue
-                if field.is_relation and (field.many_to_many or field.one_to_many):
+                if field.is_relation and field.one_to_many:
                     continue
                 fields.append(field.name)
             # TODO: apply a default ordering "smarter" than declaration order? Alphabetical? By field type?
             # TODO: allow model to specify an alternative field ordering?
 
         data = {}
+
+        if isinstance(instance, TreeModel):
+            # using `_hierarchy` with the prepended `_` to try to archive a unique name, in cases where a model might have hierarchy field.
+            data["_hierarchy"] = instance
```

Comment:
Why does this need to be done here instead of handled in `render_value`?

Reply:
Because it needs to be part of the `data`, if not it would not be seen by `render_value` because it is not a model field